### PR TITLE
Source Hubspot: ensure all oauth2.0 scopes in "check" command

### LINKED ISSUE #12484: Source Hubspot: 403 error on sync
[A common problem users have](https://github.com/airbytehq/alpha-beta-issues/issues/41) with the Hubspot connector is they haven't authorized the required scopes.

While this is a configuration issue, not a system error, this is happening frequently enough that we should improve the flow.

## Possible solutions
1. The check method could try reading data from all the streams to ensure the right permissions are set (we currently only check one)
2. We could check the authorized scopes and not sync the streams that aren't permitted (similar to what we do for the [shopify source](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-shopify/source_shopify/source.py#L542)
3. We could catch 403 errors and raise an exception with more details about how the user can troubleshoot the permission issues

Since this is derived from an OC issue we want to resolve, I would favor solution 1.

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
Check that oauth2.0 token has all needed scopes

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [x] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]
Looks nice, however we're not testing the actual data fetch using the granted scopes. Is it possible that access_token has the right granted scopes, but the user still cannot fetch the data due to role or permission limitations? 
The analogy is `source salesforce`, where you can have all the scopes granted, but due to data object limitations, the data could not be fetched until the user assigned to that object or the user's role gives such an opportunity.

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py [hunk 1]
```diff
@@ -44,22 +45,60 @@
...
+    "sales-email-read",
+    "tickets",
+]
+
 
 class SourceHubspot(AbstractSource):
     def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
         """Check connection"""
+        common_params = self.get_common_params(config=config)
+        if common_params.get("authenticator"):
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:74]: shouldn't we do the same in `def streams(self, config: Mapping[str, Any]) -> List[Stream]:`? My suggestion is to move this code to `def get_common_params` and probably rename it

**Review by davydov-d — CHANGES_REQUESTED** [review]

**Comment by grubberr** [comment]
@bazarnov I did not saw such cases like for `salesforce`,
where some data not accessible even if we have needed scopes,
but I will continue to watch.

**Review by girarda — APPROVED** [review]
looks good! Approved pending updating as per denis' suggestion

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py (hunk 1 above)
Comment (grubberr) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:74]: I understand your point  you want to make this token checking in another command (not only "check" command)

As for me we don't need to do it
we have exectly one place for checking - "check" command

In subsequent commands we just use token as is

Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:74]: will it be the same token? I mean `get_access_token()` can return different tokens from one call(check) to another(discover), right?
can the token be edited - granted additional scopes or revoked some?

Comment (grubberr) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:74]: Majority of Oauth2.0 API(s) grant `scopes` when you get refresh_token

Maybe thare are APIs which allows to downgrade scopes on
refresh_token -> access_token exchanged 
but I did not see such API(s)

Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:74]: Okay then

**Review by davydov-d — APPROVED** [review]

**Comment by shrodingers** [comment]
Hello ! Just saw that this has been merged on the last version, and was wondering how you suggest to proceed when some scopes that are now required are not available for an organization ? It blocks to sync just a subset of streams, if there are some scopes that are not available but needed for other streams.

**Pull request merged**